scripts/thaliPrerequisite.js

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+/*
+ This script will do the following tasks
+ 0. Clone the working jxcore-cordova plugin
+ 1. Fix issue on "can not replace existing file"
+ 2. Install the thali node module to global (until we get thali in npm repo)
+ 3. Move the build-extras.gradle to platform
+ 4: Create a dummy file to prevent future execution by other plugin add operation
+ */
+
+
+var fs = require('fs');
+var exec = require('child_process').execSync;
+
+
+var rootdir = process.argv[2];
+//Dummy file to prevent the execution of this script when the developer add other Cordova plugins
+var dummyFile = rootdir + '/../plugins/org.thaliproject.p2p/dummy.js';
+
+//Replace JXcoreExtension.java
+//Copy from \plugins\org.thaliproject.p2p\src\android\java\io\jxcore\node to \platforms\android\src\io\jxcore\node
+function replaceJXCoreExtension(){
+    var sourceFile =  rootdir + '/../plugins/org.thaliproject.p2p/src/android/java/io/jxcore/node/JXcoreExtension.java';
+    var targetFile = rootdir + '/../platforms/android/src/io/jxcore/node/JXcoreExtension.java';
+    fs.writeFileSync(targetFile, fs.readFileSync(sourceFile));
```

Comment:
There are several issues with line 25.
1. ReadFileSync reads the entire file into memory. Now in this case it really doesn't matter because the file is so small but in general it's not a good approach. You really want to use something like createReadStream which will allow the contents to be streamed instead of read in memory and then moved over.
2. As shown in the example [here](http://stackoverflow.com/questions/11293857/fastest-way-to-copy-file-in-node-js) (check the first answer) you really want to use pipe. It's more idiomatic.
3. As mentioned in the first comment in the response in the link above, you absolutely have to deal with errors. On both the createReadStream and createWriteStream you need to have callbacks to catch errors, log them and then throw the error so the install process fails and the user can try to figure out what went wrong.